daemon/nftables.c

```diff
@@ -518,27 +518,31 @@ static const char *delete_chain(struct mnl_socket *nl, int family, uint32_t *seq
 static const char *nftables_shutdown_family(struct mnl_socket *nl, int family, uint32_t *seq,
 		const char *chain, const char *base_chain, nftables_args *dummy)
 {
-	// clean up rules in legacy `INPUT` chain
-	const char *err = iterate_rules(nl, family, "INPUT", seq,
-			&(struct iterate_callbacks) {
-				.parse_expr = match_immediate_rtpe,
-				.chain = chain,
-				.rule_final = check_immediate,
-				.iterate_final = iterate_delete_rules,
-			});
-	if (err)
-		return err;
+	const char *err;
 
-	// clean up rules in `input` chain
-	err = iterate_rules(nl, family, "input", seq,
-			&(struct iterate_callbacks) {
-				.parse_expr = match_immediate_rtpe,
-				.chain = chain,
-				.rule_final = check_immediate,
-				.iterate_final = iterate_delete_rules,
-			});
-	if (err)
-		return err;
+	if (!base_chain || strcmp(base_chain, "none")) {
+		// clean up rules in legacy `INPUT` chain
+		err = iterate_rules(nl, family, "INPUT", seq,
+				&(struct iterate_callbacks) {
+					.parse_expr = match_immediate_rtpe,
+					.chain = chain,
+					.rule_final = check_immediate,
+					.iterate_final = iterate_delete_rules,
+				});
+		if (err)
```

Comment:
Does this commit fix a particular issue that you're seeing? Because this was done like this on purpose, to make transitioning from the previous iptables-extension package easier and automatic.

Reply:
Yes. I have a custom rule in the INPUT chain to jump to the rtpengine chain. This jump rule is removed at rtpengine shutdown **and startup**.

This change should only affect configs with `nftables-base-chain = none`. The default is not "none", so it should not affect people transitioning.

Documentation for  `nftables-base-chain` says:
> If this option is set to the special string none, then rtpengine will create its custom chain and rule as it normally would, but will skip adding an immediate-goto rule to the custom chain. Doing so requires the operator to manually create this immediate-goto rule somewhere themselves.

This won't work if rtpengine deletes said rule.

Reply:
> Yes. I have a custom rule in the INPUT chain to jump to the rtpengine chain. This jump rule is removed at rtpengine shutdown **and startup**.
> 
> This change should only affect configs with `nftables-base-chain = none`. The default is not "none", so it should not affect people transitioning.
> 
> Documentation for `nftables-base-chain` says:
> 
> > If this option is set to the special string none, then rtpengine will create its custom chain and rule as it normally would, but will skip adding an immediate-goto rule to the custom chain. Doing so requires the operator to manually create this immediate-goto rule somewhere themselves.
> 
> This won't work if rtpengine deletes said rule. My config:
> 
> ```
> nftables-chain = rtpengine-0
> nftables-base-chain = none
> nftables-family = ip
> ```
> 
> Also, IMHO rtpengine should not touch and/or remove rules and/or chains it did not add itself.

Alright, makes sense, thanks